### ISSUE TITLE
TLS auth not supported for https lookup in C++ client

### ISSUE THREAD

**Issue by ivankelly** [apache/pulsar#2314]
Try the following test:
```
TEST(AuthPluginTest, testTlsDetectHttps) {
    ClientConfiguration config = ClientConfiguration();
    config.setUseTls(true); // shouldn't be needed soon
    config.setTlsTrustCertsFilePath("../../pulsar-broker/src/test/resources/authentication/tls/cacert.pem");
    config.setTlsAllowInsecureConnection(false);
    AuthenticationPtr auth =
        pulsar::AuthTls::create("../../pulsar-broker/src/test/resources/authentication/tls/client-cert.pem",
                                "../../pulsar-broker/src/test/resources/authentication/tls/client-key.pem");
    config.setAuth(auth);

    Client client("https://localhost:9766", config);

    std::string topicName = "persistent://property/cluster/namespace/test-tls-detect-https";

    Producer producer;
    Promise<Result, Producer> producerPromise;
    client.createProducerAsync(topicName, WaitForCallbackValue<Producer>(producerPromise));
    Future<Result, Producer> producerFuture = producerPromise.getFuture();
    Result result = producerFuture.get(producer);
    ASSERT_EQ(ResultOk, result);
}
```

Should work. Does not. Code isn't there to support it.

**Comment by merlimat** [comment]
Is it just trying to use HTTP?

**Comment by ivankelly** [comment]
not sure. but i checked the connection code, and it doesn't pull any tls auth stuff

**Comment by sijie** [comment]
this is fixed by #2350
